src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -63,41 +68,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     }
 
     String query = encodeValue(param);
-    List<String> allLinks = new ArrayList<>();
-
-    // TODO: after implementing scraping, consider changing getLink calls to a
-    // for-loop
-
-    /*
-     * Send request to retrieve card content through w3School site link from Google
-     * CSE
-     */
-    String w3Link = getLink(W3_CSE_ID, query);
-    if (w3Link != null) {
-      allLinks.add(w3Link);
-      // TODO: Call scraping function to return JSON card content
-    }
-
-    /*
-     * Send request to retrieve card content through StackOverflow site link from
-     * Google CSE
-     */
-    String stackLink = getLink(STACK_CSE_ID, query);
-    if (stackLink != null) {
-      allLinks.add(stackLink);
-      // TODO: Call stackoverflow API to return JSON card content
-    }
-
-    /*
-     * Send request to retrieve card content through GeeksForGeeks site link from
-     * Google CSE
-     */
-    String geeksLink = getLink(GEEKS_CSE_ID, query);
-    if (geeksLink != null) {
-      allLinks.add(geeksLink);
-      // TODO: Call scraping function to return JSON card content
+    List<Card> allCards = new ArrayList<>();
+    List<Client> clients = new ArrayList<>();
+    clients.add(new W3SchoolClient(W3_CSE_ID));
```

Comment:
I know this part of the card is volatile since 2 other in-progress PRs are touching it, but this list ought to be computed in the constructor instead of on every request.